provider/core/src/serde.rs

```diff
@@ -216,32 +217,12 @@ where
 }
 
 /// A wrapper around `&dyn `[`SerdeSeDataStruct`]`<'data>` for integration with DataProvider.
-pub struct SerdeSeDataStructWrap<'b, 'data> {
-    inner: &'b (dyn SerdeSeDataStruct<'data> + 'data),
-}
+pub struct SerdeSeDataStructWrap<'data>(&'data (dyn SerdeSeDataStruct<'data> + 'data));
 
-impl<'b, 'data> Deref for SerdeSeDataStructWrap<'b, 'data> {
```

Comment:
Not sure whether it's better with two lifetimes or one lifetime on this helper struct.